remote-api/src/main/java/org/alfresco/rest/api/nodes/NodesEntityResource.java

```diff
@@ -225,5 +269,111 @@ public DirectAccessUrl requestContentDirectUrl(String nodeId, DirectAccessUrlReq
         }
         return directAccessUrl;
     }
+
+    /**
+     * Folder Size - returns size of a folder.
+     *
+     * @param nodeId String id of folder - will also accept well-known alias, eg. -root- or -my- or -shared-
+     *               Please refer to OpenAPI spec for more details !
+     * Returns the executionId which shows pending action, which can be used in a
+     * GET/calculateSize endpoint to check if the action's status has been completed, comprising the size of the node in bytes.
+     *               <p>
+     *               If NodeId does not exist, EntityNotFoundException (status 404).
+     *               If nodeId does not represent a folder, InvalidNodeTypeException (status 422).
+     */
+    @Operation("calculate-folder-size")
+    @WebApiDescription(title = "Calculating Folder Size", description = "Calculating size of a folder", successStatus = Status.STATUS_ACCEPTED)
+    @WebApiParameters({@WebApiParam(name = "nodeId", title = "The unique id of Execution Job", description = "A single nodeId")})
+    public Map<String, Object> calculateFolderSize(String nodeId, Void ignore, Parameters parameters, WithResponse withResponse)
+    {
+        NodeRef nodeRef = nodes.validateNode(nodeId);
+        int maxItems = Math.min(parameters.getPaging().getMaxItems(), 1000);
+        QName qName = nodeService.getType(nodeRef);
+        Map<String, Object> result = new HashMap<>();
+        validatePermissions(nodeRef, nodeId);
+
+        if(!FOLDER.equalsIgnoreCase(qName.getLocalName()))
+        {
+            throw new InvalidNodeTypeException(INVALID_NODEID);
+        }
+
+        try
+        {
+            FolderSizeImpl folderSizeImpl = new FolderSizeImpl(actionService);
+            return folderSizeImpl.executingAsynchronousFolderAction(nodeRef, maxItems, result, simpleCache);
+        }
+        catch (Exception alfrescoRuntimeError)
+        {
+            LOG.error("Exception occurred in NodesEntityResource:createById {}", alfrescoRuntimeError.getMessage());
+            throw new AlfrescoRuntimeException("Exception occurred in NodesEntityResource:createById",alfrescoRuntimeError);
+        }
+    }
```

Comment:
formatting issue, add line break.

Reply:
Done.

Reply:
The entire method needs to be cleaned up. As I mentioned earlier, the business logic should be implemented in the respective '...Impl' class, not directly in the resource. Additionally, the return type is incorrect.